src/cromshell/logs/command.py

```diff
@@ -0,0 +1,267 @@
+        # function on that subworkflow
+        if (
+            "subWorkflowMetadata" in workflow_metadata["calls"][task][0]
+            and expand_sub_workflows
+        ):
+            sub_workflow_name = task
+            task_shards = workflow_metadata["calls"][sub_workflow_name]
+            print(f"{indent}SubWorkflow {sub_workflow_name}")
+
+            # For each element in total number of subworkflow calls get the subworkflow
+            # metadata. This loop will go through each shard if task is scattered
+            for i in range(len(task_shards) - 1):
+                sub_workflow_metadata = task_shards[i]["subWorkflowMetadata"]
+
+                print_workflow_logs(
+                    workflow_metadata=sub_workflow_metadata,
+                    indent=indent + "\t",
+                    expand_sub_workflows=expand_sub_workflows,
+                    status_keys=status_keys,
+                    cat_logs=cat_logs,
+                )
+
+        # If no subworkflow is found then print status summary for the task
+        else:
+            did_print = (
+                print_task_logs(task, indent, workflow_metadata, status_keys, cat_logs)
+                or did_print
+            )
+
+    return did_print
+
+
+def print_task_logs(
+    task: str,
+    indent: str,
+    workflow_metadata: dict,
+    status_keys: list,
+    cat_logs: bool,
+) -> bool:
+    """
+    Prints the backend logs from the workflow
+    :param task: Name of the task
+    :param indent: Indent string given as "\t", used to indent print out
+    :param workflow_metadata: Metadata of the workflow to process
+    :param cat_logs: Will use GCS to attempt to print the logs
+    :return: true if any logs were printed
+    """
+
+    did_print = False
+
+    shard_list = workflow_metadata["calls"][task]
+
+    sharded = workflow_metadata["calls"][task][0]["shardIndex"] != -1
+
+    for i in range(len(shard_list)):
+
+        status = shard_list[i]["executionStatus"]
+        if "ALL" in status_keys or status in status_keys:
+            if "Done" in status:
+                task_status_font = io_utils.TextStatusesColor.TASK_COLOR_SUCCEEDED
+            elif "Running" in status:
+                task_status_font = io_utils.TextStatusesColor.TASK_COLOR_RUNNING
+            elif "RetryableFailure" in status:
+                task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILING
+            elif "Failed":
+                task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILED
+            else:
+                task_status_font = None
+
+            shardstring = (
+                "" if not sharded else "-shard-" + str(shard_list[i]["shardIndex"])
+            )
+            logs = shard_list[i]["backendLogs"]["log"]
+            if cat_logs:
+                print(
+                    colored(
+                        f"\n\n\n{'='*90}\n{indent}{task}{shardstring}:\t{status}\t {logs}\n{'='*90}",
```

Comment:
Lets replace this hardcoded 90 with a dynamic padding based on `os.get_terminal_size()`